File: javalibrary/src/main/java/com/bluthlee/javalibrary/ui/RecorderWidgetFactory.java
package com.bluthlee.javalibrary.ui;

import android.content.Context;

import com.bluthlee.javalibrary.data.DataBinder;
import com.bluthlee.javalibrary.ui.recorderWidget.DragRecorderWidget;
import com.bluthlee.javalibrary.ui.recorderWidget.FloatRecorderWidget;
import com.bluthlee.javalibrary.ui.recorderWidget.RecorderWidget;

/**
 * Created by LC on 2017/8/31.
 */

public class RecorderWidgetFactory {

    public static RecorderWidget getRecorderWidget(int style, Context context, DataBinder dataBinder) {
        switch (style) {
            case DataBinder.STYLE_DRAG:
                return new DragRecorderWidget(dataBinder);
            case DataBinder.STYLE_FLOAT:
                return new FloatRecorderWidget(context, dataBinder);
        }
        return null;
    }

}

File: kotlinlibrary/src/main/java/com/bluthlee/library/ui/RecorderWidgetFactory.kt
package com.bluthlee.library.ui

import android.content.Context
import com.bluthlee.library.data.DataBinder
import com.bluthlee.library.ui.recorderWidget.DragRecorderWidget
import com.bluthlee.library.ui.recorderWidget.FloatRecorderWidget
import com.bluthlee.library.ui.recorderWidget.RecorderWidget

/**
 * Created by LC on 2017/8/24.
 */
internal class RecorderWidgetFactory {

    companion object {

        fun getRecorderWidget(style: Int, context: Context, dataBinder: DataBinder): RecorderWidget? {
            when (style) {
                DataBinder.STYLE_DRAG -> {
                    return DragRecorderWidget(dataBinder)
                }
                DataBinder.STYLE_FLOAT -> {
                    return FloatRecorderWidget(context, dataBinder)
                }
            }
            return null
        }

    }

}